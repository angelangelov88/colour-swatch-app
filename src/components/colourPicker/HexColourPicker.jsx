import React, { useState } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { HexColorInput, HexColorPicker } from 'react-colorful';
import './ColourPicker.css';
import { DuplicateIcon } from '@heroicons/react/outline';
// import { Tooltip } from '../tooltip/Tooltip';
// import { HexColourInput } from './HexColourInput';

function ColourPicker(props) {
  const {
    children,
  } = props;

  const [color, setColor] = useState('#ffffff');

  function adjustColour(mainColor, amount) {
    // eslint-disable-next-line no-shadow
    return `#${mainColor.replace(/^#/, '').replace(/../g, (mainColor) => (`0${Math.min(255, Math.max(0, parseInt(mainColor, 16) + amount)).toString(16)}`).substr(-2))}`;
  }

  const borderColour = adjustColour(color, -50);

  const colourSquareClasses = classNames(
    'w-5 h-5 mr-2',
    'border',
    'rounded',
  );

  const colourSquareContainerClasses = classNames(
    'w-32 h-9 p-2 flex items-center group rounded-lg cursor-pointer bg-gray-10',
    '',
  );

  return (
    <div className={colourSquareContainerClasses}>
      <div className={colourSquareClasses} style={{ backgroundColor: `${color}`, borderColor: `${borderColour}` }}>
        <div className="relative shadow-lg">
          <div className="block group-hover:block absolute -top-40 left-32 p-2 rounded-lg shadow-lg">
            <HexColorPicker color={color} onChange={setColor} className=" custom-pointers example" />
            <div className="flex justify-evenly items-center">
              <h1 className="text-sm font-ibm font-medium">Hex:</h1>
              <div className={colourSquareContainerClasses}>
                <div className="flex justify-evenly">
                  <div className={colourSquareClasses} style={{ backgroundColor: `${color}`, borderColor: `${borderColour}` }} />
                  <p>#</p>
                  <HexColorInput color={color} onChange={setColor} className="w-16 bg-gray-10 -mt-[3px] uppercase" />
                </div>
              </div>
              <DuplicateIcon className="w-5 h-5 text-gray-80" />
            </div>

          </div>
          {children}
        </div>
      </div>
      <div className="uppercase">
        {color}
      </div>
      <div />
    </div>

  );
}

ColourPicker.defaultProps = {
  children: undefined,
};

ColourPicker.propTypes = {
  children: PropTypes.node,
};

export default ColourPicker;
export { ColourPicker };
