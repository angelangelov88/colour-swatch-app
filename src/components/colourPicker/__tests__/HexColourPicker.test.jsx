import React from 'react';
import renderer from 'react-test-renderer';
import HexColourPicker from '../HexColourPicker';

it('should render with defaults', () => {
  const component = renderer.create(
    <HexColourPicker />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with initial color set to #406024', () => {
  const component = renderer.create(
    <HexColourPicker
      colourValue="#406024"
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with specified position', () => {
  const component = renderer.create(
    <HexColourPicker
      position="top-9 left-0"
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
