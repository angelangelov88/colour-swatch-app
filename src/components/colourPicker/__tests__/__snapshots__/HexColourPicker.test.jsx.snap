// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with defaults 1`] = `
<div
  className="w-32 h-9 p-2 flex items-center group rounded-lg cursor-pointer bg-gray-10 font-ibm text-sm relative"
>
  <div
    className="w-5 h-5 mr-2 border rounded"
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#cdcdcd",
      }
    }
  />
  <div
    className="uppercase"
  >
    #ffffff
  </div>
  <div
    className="hidden group-hover:block absolute p-2 rounded-lg shadow-lg -top-40 left-[120px]"
  >
    <div
      className="react-colorful  custom-pointers example"
    >
      <div
        className="react-colorful__saturation"
        style={
          Object {
            "backgroundColor": "hsl(0, 100%, 50%)",
          }
        }
      >
        <div
          aria-label="Color"
          aria-valuetext="Saturation 0%, Brightness 100%"
          className="react-colorful__interactive"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
        >
          <div
            className="react-colorful__pointer react-colorful__saturation-pointer"
            style={
              Object {
                "left": "0%",
                "top": "0%",
              }
            }
          >
            <div
              className="react-colorful__pointer-fill"
              style={
                Object {
                  "backgroundColor": "hsl(0, 0%, 100%)",
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        className="react-colorful__hue react-colorful__last-control"
      >
        <div
          aria-label="Hue"
          aria-valuetext={0}
          className="react-colorful__interactive"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
        >
          <div
            className="react-colorful__pointer react-colorful__hue-pointer"
            style={
              Object {
                "left": "0%",
                "top": "50%",
              }
            }
          >
            <div
              className="react-colorful__pointer-fill"
              style={
                Object {
                  "backgroundColor": "hsl(0, 100%, 50%)",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="flex justify-evenly items-center"
    >
      <div
        className="font-medium"
      >
        Hex:
      </div>
      <div
        className="w-32 h-9 p-2 flex items-center group rounded-lg cursor-pointer bg-gray-10 font-ibm text-sm relative"
      >
        <div
          className="flex justify-evenly"
        >
          <div
            className="w-5 h-5 mr-2 border rounded"
            style={
              Object {
                "backgroundColor": "#ffffff",
                "borderColor": "#cdcdcd",
              }
            }
          />
          <p>
            #
          </p>
          <input
            className="w-16 bg-gray-10 -mt-[3px] uppercase focus:outline-none"
            onBlur={[Function]}
            onChange={[Function]}
            spellCheck="false"
            value="ffffff"
          />
        </div>
      </div>
      <div
        className="w-8 h-8 border rounded-lg shadow-sm p-1"
      >
        <svg
          aria-hidden="true"
          className="w-6 h-6 text-gray-80 "
          fill="none"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth={2}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 16H6a2 2 0 01-2-2V6a2 2 0 012-2h8a2 2 0 012 2v2m-6 12h8a2 2 0 002-2v-8a2 2 0 00-2-2h-8a2 2 0 00-2 2v8a2 2 0 002 2z"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render with initial color set to #406024 1`] = `
<div
  className="w-32 h-9 p-2 flex items-center group rounded-lg cursor-pointer bg-gray-10 font-ibm text-sm relative"
>
  <div
    className="w-5 h-5 mr-2 border rounded"
    style={
      Object {
        "backgroundColor": "#406024",
        "borderColor": "#0e2e00",
      }
    }
  />
  <div
    className="uppercase"
  >
    #406024
  </div>
  <div
    className="hidden group-hover:block absolute p-2 rounded-lg shadow-lg -top-40 left-[120px]"
  >
    <div
      className="react-colorful  custom-pointers example"
    >
      <div
        className="react-colorful__saturation"
        style={
          Object {
            "backgroundColor": "hsl(92, 100%, 50%)",
          }
        }
      >
        <div
          aria-label="Color"
          aria-valuetext="Saturation 63%, Brightness 38%"
          className="react-colorful__interactive"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
        >
          <div
            className="react-colorful__pointer react-colorful__saturation-pointer"
            style={
              Object {
                "left": "63%",
                "top": "62%",
              }
            }
          >
            <div
              className="react-colorful__pointer-fill"
              style={
                Object {
                  "backgroundColor": "hsl(92, 46%, 26%)",
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        className="react-colorful__hue react-colorful__last-control"
      >
        <div
          aria-label="Hue"
          aria-valuetext={92}
          className="react-colorful__interactive"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
        >
          <div
            className="react-colorful__pointer react-colorful__hue-pointer"
            style={
              Object {
                "left": "25.555555555555554%",
                "top": "50%",
              }
            }
          >
            <div
              className="react-colorful__pointer-fill"
              style={
                Object {
                  "backgroundColor": "hsl(92, 100%, 50%)",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="flex justify-evenly items-center"
    >
      <div
        className="font-medium"
      >
        Hex:
      </div>
      <div
        className="w-32 h-9 p-2 flex items-center group rounded-lg cursor-pointer bg-gray-10 font-ibm text-sm relative"
      >
        <div
          className="flex justify-evenly"
        >
          <div
            className="w-5 h-5 mr-2 border rounded"
            style={
              Object {
                "backgroundColor": "#406024",
                "borderColor": "#0e2e00",
              }
            }
          />
          <p>
            #
          </p>
          <input
            className="w-16 bg-gray-10 -mt-[3px] uppercase focus:outline-none"
            onBlur={[Function]}
            onChange={[Function]}
            spellCheck="false"
            value="406024"
          />
        </div>
      </div>
      <div
        className="w-8 h-8 border rounded-lg shadow-sm p-1"
      >
        <svg
          aria-hidden="true"
          className="w-6 h-6 text-gray-80 "
          fill="none"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth={2}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 16H6a2 2 0 01-2-2V6a2 2 0 012-2h8a2 2 0 012 2v2m-6 12h8a2 2 0 002-2v-8a2 2 0 00-2-2h-8a2 2 0 00-2 2v8a2 2 0 002 2z"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render with specified position 1`] = `
<div
  className="w-32 h-9 p-2 flex items-center group rounded-lg cursor-pointer bg-gray-10 font-ibm text-sm relative"
>
  <div
    className="w-5 h-5 mr-2 border rounded"
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#cdcdcd",
      }
    }
  />
  <div
    className="uppercase"
  >
    #ffffff
  </div>
  <div
    className="hidden group-hover:block absolute p-2 rounded-lg shadow-lg top-9 left-0"
  >
    <div
      className="react-colorful  custom-pointers example"
    >
      <div
        className="react-colorful__saturation"
        style={
          Object {
            "backgroundColor": "hsl(0, 100%, 50%)",
          }
        }
      >
        <div
          aria-label="Color"
          aria-valuetext="Saturation 0%, Brightness 100%"
          className="react-colorful__interactive"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
        >
          <div
            className="react-colorful__pointer react-colorful__saturation-pointer"
            style={
              Object {
                "left": "0%",
                "top": "0%",
              }
            }
          >
            <div
              className="react-colorful__pointer-fill"
              style={
                Object {
                  "backgroundColor": "hsl(0, 0%, 100%)",
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        className="react-colorful__hue react-colorful__last-control"
      >
        <div
          aria-label="Hue"
          aria-valuetext={0}
          className="react-colorful__interactive"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
        >
          <div
            className="react-colorful__pointer react-colorful__hue-pointer"
            style={
              Object {
                "left": "0%",
                "top": "50%",
              }
            }
          >
            <div
              className="react-colorful__pointer-fill"
              style={
                Object {
                  "backgroundColor": "hsl(0, 100%, 50%)",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="flex justify-evenly items-center"
    >
      <div
        className="font-medium"
      >
        Hex:
      </div>
      <div
        className="w-32 h-9 p-2 flex items-center group rounded-lg cursor-pointer bg-gray-10 font-ibm text-sm relative"
      >
        <div
          className="flex justify-evenly"
        >
          <div
            className="w-5 h-5 mr-2 border rounded"
            style={
              Object {
                "backgroundColor": "#ffffff",
                "borderColor": "#cdcdcd",
              }
            }
          />
          <p>
            #
          </p>
          <input
            className="w-16 bg-gray-10 -mt-[3px] uppercase focus:outline-none"
            onBlur={[Function]}
            onChange={[Function]}
            spellCheck="false"
            value="ffffff"
          />
        </div>
      </div>
      <div
        className="w-8 h-8 border rounded-lg shadow-sm p-1"
      >
        <svg
          aria-hidden="true"
          className="w-6 h-6 text-gray-80 "
          fill="none"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth={2}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 16H6a2 2 0 01-2-2V6a2 2 0 012-2h8a2 2 0 012 2v2m-6 12h8a2 2 0 002-2v-8a2 2 0 00-2-2h-8a2 2 0 00-2 2v8a2 2 0 002 2z"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
